applications/StructuralMechanicsApplication/custom_elements/updated_lagrangian.cpp

```diff
@@ -0,0 +1,565 @@
+        
+        if (rThisKinematicVariables.detJ0 < 0.0)
+        {
+            KRATOS_ERROR << "WARNING:: ELEMENT ID: " << this->Id() << " INVERTED. DETJ0: " << rThisKinematicVariables.detJ0 << std::endl;
+        }
+        
+        // Deformation gradient
+        const unsigned int strain_size = (rThisKinematicVariables.B).size1();
+        noalias( rThisKinematicVariables.F ) = prod( rThisKinematicVariables.J0, rThisKinematicVariables.InvJ );
+//         Matrix delta_displacement = CalculateDeltaDisplacement();
+//         CalculateF(rThisKinematicVariables.F, delta_displacement,rThisKinematicVariables.DN_DX,strain_size,IntegrationPoints,PointNumber);
+        
+        // Calculating operator B
+        CalculateB( rThisKinematicVariables.B, rThisKinematicVariables.DN_DX, strain_size, IntegrationPoints, PointNumber );
+    }
+
+    //************************************************************************************
+    //************************************************************************************
+    
+    void UpdatedLagrangian::CalculateConstitutiveVariables(
+        KinematicVariables& rThisKinematicVariables, 
+        ConstitutiveVariables& rThisConstitutiveVariables, 
+        ConstitutiveLaw::Parameters& rValues,
+        const unsigned int PointNumber,
+        const GeometryType::IntegrationPointsArrayType& IntegrationPoints,
+        const ConstitutiveLaw::StressMeasure ThisStressMeasure,
+        const Vector Displacements
+        )
+    {
+        // Axisymmetric case
+        if (rThisConstitutiveVariables.StrainVector.size() == 4)
+        {
+            rThisKinematicVariables.F.resize(3, 3); // We keep the old values
+            for (unsigned int index = 0; index < 1; index++)
+            {
+                rThisKinematicVariables.F(index, 2) = 0.0;
+                rThisKinematicVariables.F(2, index) = 0.0;
+            }
+
+            rThisKinematicVariables.N = GetGeometry().ShapeFunctionsValues( rThisKinematicVariables.N, IntegrationPoints[PointNumber].Coordinates() );
+            const double current_radius = StructuralMechanicsMathUtilities::CalculateRadius(rThisKinematicVariables.N, GetGeometry(), Current);
+            const double initial_radius = StructuralMechanicsMathUtilities::CalculateRadius(rThisKinematicVariables.N, GetGeometry(), Initial);
+            rThisKinematicVariables.F(2, 2) = current_radius/initial_radius;
+        }
+        
+        // Here we essentially set the input parameters
+        rThisKinematicVariables.detF = MathUtils<double>::Det(rThisKinematicVariables.F);
+        const double detFT = rThisKinematicVariables.detF * mDetF0[PointNumber];
+        rValues.SetDeterminantF(detFT); //assuming the determinant is computed somewhere else
+        const Matrix FT = prod(rThisKinematicVariables.F, mF0[PointNumber]);
+        rValues.SetDeformationGradientF(FT); //F computed somewhere else
```

Comment:
IMPORTANT: rThisKinematicVariables.F should contain Dxn+1/dX0

from this code, i understant that you are using it to store DF:=Dx(n+1)/Dx(n)

please do not do it since it is inconsistent with what done in the other elements.

if you need DF store it as another variable